modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy

```diff
@@ -348,6 +358,17 @@ class CmdRun extends CmdBase implements HubOptions {
         runner.execute(scriptArgs, this.entryName)
     }
 
+    protected Collection checkNxfEnv(ConfigMap config) {
+        // Warn about setting NXF_ environment variables within env config scope
+        final nxfEnvs = config.env.findAll { it ->
+            it.toString().startsWith('NXF_') &&
+                    ! it.toString().startsWith('NXF_DEBUG')
+        }
+        nxfEnvs.collect {
+            it.toString().split('=')[0]
```

Comment:
nxfEnvs is a map, probably you can return something as

```
nxfEnvs.collect{ it.key }
```

Reply:
It should, but for some reason it does not 😕.

nxfEnvs is an ArrayList. When I run this
`config.env.findAll { log.warn it.key as String }`
I get `[Static type checking] - No such property: key for class: java.lang.Object`, even though `config.env.getClass()` returns `configMap`.

Reply:
Now it all makes sense, Jorge! Thanks for the help.